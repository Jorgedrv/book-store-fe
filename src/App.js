import React from 'react';

const App = () => {
    return <div>My first app, let's staterd</div>;
};

export default App;