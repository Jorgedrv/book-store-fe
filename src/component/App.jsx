import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Layout from './layout/Layout';
import BookContainer from './book/BookContainer';
import Login from './user/Login';
import Register from './user/Register';
import Auth from './auth/index';
import { SnackbarProvider } from 'notistack';

const App = () => {
    return (
        <SnackbarProvider maxSnack={3}>
            <Router>
                <Layout>
                    <Switch>
                        <Route path='/login'>
                            <Login />
                        </Route>
                        <Route path='/register'>
                            <Register />
                        </Route>
                        <Route exact path='/'>
                            <Auth>
                                <BookContainer />
                            </Auth>
                        </Route>
                    </Switch>
                </Layout>
            </Router>
        </SnackbarProvider>
    );
};

export default App;