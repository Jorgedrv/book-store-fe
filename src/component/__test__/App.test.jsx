import React from 'react';
import renderWithRedux from '../../util/testUtil';
import App from '../App';

jest.mock('../book/BookContainer', () => (props) => <div {...props} />);
describe('App component', () => {
    it('App test should render with error', () => {
        const { asFragment } = renderWithRedux(<App />, {});
        expect(asFragment()).toMatchSnapshot();
    });
});
