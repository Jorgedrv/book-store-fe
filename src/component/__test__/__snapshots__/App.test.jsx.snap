// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App test should render with error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Book Store
          </h6>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <form
        autocomplete="off"
        novalidate=""
      >
        <div
          class="MuiBox-root MuiBox-root-7 makeStyles-wrapper-4"
        >
          <div
            class="MuiPaper-root makeStyles-paper-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4"
            >
              Book store Login
            </h4>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-topMargin-6"
              data-testid="email-testid"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="email"
                id="email-label"
              >
                Enter email address
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  placeholder="Enter email address"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-10"
                  >
                    <span>
                      Enter email address
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-topMargin-6"
              data-testid="password-testid"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Enter password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  placeholder="Enter password"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-10"
                  >
                    <span>
                      Enter password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-topMargin-6 MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Login
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
